nixos/tests/zfs.nix

```diff
@@ -60,6 +76,8 @@ let
             "zfs create -o mountpoint=legacy rpool/root",
             "mount -t zfs rpool/root /tmp/mnt",
             "udevadm settle",
+            "zfs set sharesmb=on rpool/root",
+            "smbclient -NL localhost",
```

Comment:
If I add `net usershare info` it says https://discourse.nixos.org/t/help-with-samba-usershares-are-currently-disabled/4817
Apart from that it seems working. Do you have anything to improve the test here?

Reply:
Tipp: You can run `$(nix-build nixos/tests/zfs.nix -A stable.driver --builders '')/bin/nixos-test-driver` to get an interactive python shell, where you can copy and paste the `machine.succeed` command

Reply:
> If I add `net usershare info` it says https://discourse.nixos.org/t/help-with-samba-usershares-are-currently-disabled/4817

Probably because samba was not started yet. I've added a command to wait for (all) services to start (or fail). But `net usershare` and `smbclient` are somewhat redundant for our purpose. Would you like to keep both checks?

Reply:
> If I add `net usershare info` it says https://discourse.nixos.org/t/help-with-samba-usershares-are-currently-disabled/4817

Probably because samba was not started yet. I've added a command to wait for all services to start (or fail). But `net usershare` and `smbclient` are somewhat redundant for our purporse. Would you like to keep both checks?

Reply:
No. Just use the more appropriate one.

Reply:
Ok, it's already done.